File: tests/e2e/scripts/iife.ts
function main(): void {
    // Simple arrow IIFE
    const x: number = ((n: number): number => n * 2)(21);
    console.log("double:", x);

    // Arrow IIFE with multiple args
    const sum: number = ((a: number, b: number): number => a + b)(10, 20);
    console.log("sum:", sum);

    // Arrow IIFE returning string
    const msg: string = ((): string => "hello iife")();
    console.log("msg:", msg);
}

File: tests/e2e/scripts/iife.rs
fn main() {
    let x: f64 = {
        let __iife = |n: f64| -> f64 { n * 2.0 };
        __iife(21.0)
    };
    println!("{} {}", "double:", x);
    let sum: f64 = {
        let __iife = |a: f64, b: f64| -> f64 { a + b };
        __iife(10.0, 20.0)
    };
    println!("{} {}", "sum:", sum);
    let msg: String = {
        let __iife = || -> String { "hello iife" };
        __iife()
    };
    println!("{} {}", "msg:", msg);
}
